docs/pnpm.md

```diff
@@ -258,8 +260,19 @@ npm packages have "lifecycle scripts" such as `postinstall` which are documented
 
 We refer to these as "lifecycle hooks".
 
-> You can disable this feature completely by setting all packages to have no hooks, using
-> `lifecycle_hooks = { "*": [] }` in `npm_translate_lock`.
+Determining which packages have lifecycle hooks is determined by, in priority order:
+
+1. The workspace `package.json` [`pnpm.onlyBuiltDependencies` attribute](https://pnpm.io/package_json#pnpmonlybuiltdependencies) if present.
+2. The pnpm-lock.yaml `requiresBuild` attribute. Note this is only available in pnpm _before_ v9, see [pnpm #7707](https://github.com/pnpm/pnpm/issues/7707) for reasons why this attribute was removed.
+
+If the `pnpm.onlyBuiltDependencies` attribute is set, even when empty, it is considered to be the full list of
+all packages containing lifecycle hooks. This is recommended going forward for pnpm v9+ compatibility.
+
+When a package has lifecycle hooks the `lifecycle_*` attributes are applied to filter which hooks are run and how they are run.
+
+For example, you can restrict lifecycle hooks across all packages to only run `postinstall`:
+
+> `lifecycle_hooks = { "*": ["postinstall"] }` in `npm_translate_lock`.
```

Comment:
Hmm. Should we error out if you try to specify a lifecycle_hooks entry for a package that is not in an explicity `onlyBuiltDependencies` or does not have a `requiresBuilds` (pre-v9)?

For example, you set

```
lifecycle_hooks = { "foobar": ["postinstall"] }
```

but `foobar` doesn't have lifecycle hooks generated?

Reply:
Talked about it offline

Reply:
Discussed and will be a followup, has been added to the 2.0 release issue: https://github.com/aspect-build/rules_js/issues/1671